dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerImportService.java

```diff
@@ -102,19 +102,28 @@ public TrackerImportReport importTracker( TrackerImportParams params )
             notifier.notify( params.getJobConfiguration(), "(" + params.getUsername() + ") Import:Start" );
         }
 
-        List<TrackerBundle> trackerBundles = preheatBundle( params, importReport );
-
-        trackerBundles = preProcessBundle( trackerBundles, importReport );
-
-        TrackerValidationReport validationReport = validateBundle( params, importReport, trackerBundles );
-
-        if ( validationReport.hasErrors() )
+        if ( TrackerImportStrategy.DELETE == params.getImportStrategy() )
         {
```

Comment:
This part of the code is one candidate to become a really unreadable one because the logic tree here is quite complex.
Maybe we can try to understand all the different parameters that will create a different execution and the find a good way to put it in the code.
I am saying that because I am changing this chunk of code in a different PR in the same way you are doing.